### PR TITLE
Enable Event Streaming from PrefectCloudEventSubscriber via CLI 

### PR DESCRIPTION
**Pull request by gabcoyne**
**Overview:**
This pull request introduces a significant enhancement to the Prefect CLI by enabling the streaming of events directly from the `PrefectCloudEventSubscriber`. This new feature allows users to monitor the flow of events in real-time through a convenient command-line interface. Alongside this primary functionality, the CLI has also been integrated with advanced event filtering options as defined in the `filters.py` module. This combination of real-time event streaming and granular filtering provides a robust toolset for users to manage and observe Prefect events effectively. Specifically we wanted to allow redirecting or logging to files in order to deliver events to log aggregators

### Example
```bash
# Stream events with optional advanced filtering
prefect events stream --format json --event-filter '{"event": {"prefix": ["prefect.task-run."]}}'
```
This command showcases the CLI's new ability to stream events, with an example of using advanced filters to narrow down to specific event names and resource IDs.

### Checklist

- [ N/A] This pull request references any related issue by including "closes `<link to issue>`"
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [ ] This pull request includes tests or only affects documentation.
- [ ] This pull request includes a label categorizing the change e.g. `maintenance`, `fix`, `feature`, `enhancement`, `docs`.
  - If you do not have permission to add a label, a maintainer will add one for you.

For documentation changes:

- [ ] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed.

For new functions or classes in the Python SDK:

- [ ] This pull request includes helpful docstrings.
- [ ] If a new Python file was added, this pull request contains a stub page in the Python SDK docs and an entry in `mkdocs.yml` navigation.

### CODE REVIEW

**Comment by EmilRex** [comment]
:hell-yeah:

File: src/prefect/cli/events.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+                filter_dict = orjson.loads(event_filter)
+                constructed_event_filter = EventFilter(
+                    **filter_dict
+                )  # Construct the filter object
+            except orjson.JSONDecodeError:
+                exit_with_error("Invalid JSON format for filter specification")
+            except AttributeError:
+                exit_with_error("Invalid filter specification")
+
+        async with PrefectCloudEventSubscriber(
```
Comment (chrisguidry) [inline src/prefect/cli/events.py:52]: This is the one that you'll change to `PrefectCloudEventSubscriber` if you want to see account-level events once https://github.com/PrefectHQ/prefect/pull/12808 is merged.  Maybe a command-line option like `--account` or something like that to choose between the two different subscribers?

File: src/prefect/cli/events.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+@events_app.command()
+async def stream(
+    format: StreamFormat = typer.Option(
+        StreamFormat.json, "--format", help="Output format (json or text)"
+    ),
+    output_file: str = typer.Option(
+        None, "--output-file", help="File to write events to"
+    ),
+    event_filter: str = typer.Option(None, "--event-filter", help="Event filter"),
```
Comment (chrisguidry) [inline src/prefect/cli/events.py:32]: Just for future reference, the filter for audit log events on the account-level stream would be:

```
--event-filter '{"related": {"role": "actor"}}'
```

**Review by chrisguidry — APPROVED** [review]

File: src/prefect/cli/events.py (hunk 2 above)
Comment (gabcoyne) [reply src/prefect/cli/events.py]: Going to add filters in later

**Review by WillRaphaelson — APPROVED** [review]

**Comment by taylor-curran** [comment]
To see audit log events in addition to workspace events run:

`prefect events stream --account`


### Work Queue Ready - Workspace Level - Orchestration Event
Event starts with `prefect.`
```JSON
{
  "occurred": "2024-05-08 14:54:59.921125+00:00",
  "event": "prefect.work-queue.ready",
  "resource": {
    "prefect.resource.id": "prefect.work-queue.b6822e8e-efe3-4769-94dd-5b4f61ff4c27",
    "prefect.resource.name": "ecs-agent-test-1-17",
    "prefect.resource.role": "work-queue"
  },
  "related": [
    {
      "prefect.resource.id": "prefect.work-pool.669eb7d2-853f-476f-93c4-cd306c5c7f6b",
      "prefect.resource.name": "default-agent-pool",
      "prefect.work-pool.type": "prefect-agent",
      "prefect.resource.role": "work-pool"
    }
  ],
  "payload": {},
  "id": "dfcdcd15-6b7e-488b-81c5-cc1f2bf51a8d",
  "follows": null
}

```

### New Team Created - Account Level - Audit Log Event
Event starts with `prefect-cloud.`
```JSON
{
  "occurred": "2024-05-08 14:55:11.247431+00:00",
  "event": "prefect-cloud.team.created",
  "resource": {
    "prefect-cloud.name": "tay-test-29",
    "prefect-cloud.description": "",
    "prefect.resource.name": "tay-test-29",
    "prefect.resource.id": "prefect-cloud.team.f7665385-27b5-4336-ada8-7f8e6d6d28e8"
  },
  "related": [
    {
      "prefect.resource.role": "actor",
      "prefect.resource.id": "prefect-cloud.actor.5de59253-3f04-4c77-80fd-da2148ee75e1",
      "prefect.resource.name": "Taylor Curran",
      "prefect-cloud.email": "redacted@redacted.com",
      "prefect-cloud.name": "Taylor Curran",
      "prefect-cloud.handle": "taylorprefectio"
    },
    {
      "prefect.resource.role": "account",
      "prefect.resource.id": "prefect-cloud.account.0ff44498-d380-4d7b-bd68-9b52da03823f",
      "prefect.resource.name": "Sales Engineering",
      "prefect-cloud.name": "Sales Engineering",
      "prefect-cloud.handle": "sales-engineering"
    }
  ],
  "payload": {},
  "id": "7d04f22a-fb4e-46c5-b912-36683df16090",
  "follows": null
}

```

**Pull request merged**